File: src/main/java/rars/venus/editors/rsyntaxtextarea/RVSyntaxScheme.java
package rars.venus.editors.rsyntaxtextarea;

import org.fife.ui.rsyntaxtextarea.Style;
import org.fife.ui.rsyntaxtextarea.SyntaxScheme;
import rars.riscv.lang.lexing.RVTokenType;

public final class RVSyntaxScheme extends SyntaxScheme {
    public RVSyntaxScheme() {
        super(false);
        final var styles = new Style[RVTokenType.values().length + SyntaxScheme.DEFAULT_NUM_TOKEN_TYPES];
        setStyles(styles);
        restoreDefaults(null);
    }
}

File: src/main/java/rars/venus/editors/rsyntaxtextarea/RVSyntaxScheme.kt
package rars.venus.editors.rsyntaxtextarea

import org.fife.ui.rsyntaxtextarea.Style
import org.fife.ui.rsyntaxtextarea.SyntaxScheme
import rars.riscv.lang.lexing.RVTokenType

class RVSyntaxScheme : SyntaxScheme(false) {
    init {
        val styles = arrayOfNulls<Style>(RVTokenType.entries.size + DEFAULT_NUM_TOKEN_TYPES)
        setStyles(styles)
        restoreDefaults(null)
    }
}
